### PR TITLE
Split the APIs into two modules with SAML auth for admin and Repose for public

### PR DESCRIPTION
**Pull request by itzg**
# Resolves

SALUS-68

# What

Enables production-ready authentication for the public (aka tenant) APIs and admin APIs. Since SAML SSO was needed for the Racker-facing admin APIs and Repose for the public/tenant facing APIs, then two distinct authentication strategies were required. As such, the cleanest solution was to separate the two APIs into distinct applications and modules.

# How

Spring SAML was introduced into the admin API module and configured for authentication. The Repose header processing filter from `salus-common` was introduced into the public API module for its authentication.

## How to test

By default, both APIs are usable as before since authentication is disabled except that the admin API now binds to port 8888 by default. The admin API can be locally tested for SAML authentication using the information in its README.

# Why

I had originally hoped to use Identity/Repose for both APIs; however, Racker authentication is only supported by the internal Identity endpoint, which is not readily routable from the public cloud hosting of the API services. As such, our hand was forced to use SAML SSO for the admin API.

# TODO

none

### CODE REVIEW

File: public/src/main/java/com/rackspace/salus/telemetry/api/config/ApiPublicProperties.java [hunk 1]
```diff
@@ -22,10 +22,13 @@
...
 @Component
 @Data
-public class ApiProperties {
-    String etcdUrl = "http://localhost:2379";
+public class ApiPublicProperties {
 
+  /**
+   * The roles (without "ROLE_" prefix) that are required to allow the user to make use of tenant APIs.
+   */
+  String[] roles = new String[]{"COMPUTE_USER"};
```
Comment (jjbuchan) [inline public/src/main/java/com/rackspace/salus/telemetry/api/config/ApiPublicProperties.java]: Hmm... this doesn't seem right.  Is this just a holder for now or is it supposed to serve a purpose?

File: admin/src/main/java/com/rackspace/salus/telemetry/api/config/AdminWebSecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+    successRedirectHandler.setDefaultTargetUrl("/admin/profile");
+    return successRedirectHandler;
+  }
+
+  @Bean
+  public SimpleUrlAuthenticationFailureHandler authenticationFailureHandler() {
+    SimpleUrlAuthenticationFailureHandler failureHandler =
+        new SimpleUrlAuthenticationFailureHandler();
+    failureHandler.setUseForward(true);
+    failureHandler.setDefaultFailureUrl("/error");
```
[the remark this answers is not in the document]
Comment (itzg) [reply admin/src/main/java/com/rackspace/salus/telemetry/api/config/AdminWebSecurityConfig.java]: Spring Boot is already registering things from their `BasicErrorController` mapped onto `/error`. For example, from logs I was able to double check this:

```
2018-12-04 08:54:10.990  INFO 96994 --- [  restartedMain] s.w.s.m.m.a.RequestMappingHandlerMapping : Mapped "{[/error]}" onto public org.springframework.http.ResponseEntity<java.util.Map<java.lang.String, java.lang.Object>> org.springframework.boot.autoconfigure.web.servlet.error.BasicErrorController.error(javax.servlet.http.HttpServletRequest)
2018-12-04 08:54:10.991  INFO 96994 --- [  restartedMain] s.w.s.m.m.a.RequestMappingHandlerMapping : Mapped "{[/error],produces=[text/html]}" onto public org.springframework.web.servlet.ModelAndView org.springframework.boot.autoconfigure.web.servlet.error.BasicErrorController.errorHtml(javax.servlet.http.HttpServletRequest,javax.servlet.http.HttpServletResponse)
```

File: admin/src/main/java/com/rackspace/salus/telemetry/api/services/SamlUserDetailsServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        .of(samlCredential.getAttributeAsStringArray("http://schemas.xmlsoap.org/claims/Group"))
+        .map(group ->
+            group
+                .replaceAll("\\W+", "_")
+                .toUpperCase()
+        )
+        .map(normalized -> new SimpleGrantedAuthority("ROLE_" + normalized))
+        .collect(Collectors.toList());
+
+    return new User(userId, "n/a",
```
[the remark this answers is not in the document]
Comment (itzg) [reply admin/src/main/java/com/rackspace/salus/telemetry/api/services/SamlUserDetailsServiceImpl.java]: Yeah, it did feel kinda wrong. I'll try a null, but I think I'd be more comfortable with just `""` as a compromise.

[the remark this answers is not in the document]
Comment (itzg) [reply admin/src/main/java/com/rackspace/salus/telemetry/api/services/SamlUserDetailsServiceImpl.java]: Things seem to expect it to be non-null, so I'll switch to `""`

https://github.com/spring-projects/spring-security/blob/master/core/src/main/java/org/springframework/security/core/userdetails/User.java#L385-L390

File: admin/pom.xml [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>com.rackspace.salus</groupId>
+            <artifactId>salus-telemetry-model</artifactId>
+            <version>0.1.0-SNAPSHOT</version>
+        </dependency>
+        <dependency>
+            <groupId>com.rackspace.salus</groupId>
+            <artifactId>salus-telemetry-label-management</artifactId>
```
[the remark this answers is not in the document]
Comment (itzg) [reply admin/pom.xml]: Thanks, it was on my mental todo list, but I had already forgotten :)

File: public/src/main/java/com/rackspace/salus/telemetry/api/config/ApiPublicProperties.java (hunk 1 above)
Comment (itzg) [reply public/src/main/java/com/rackspace/salus/telemetry/api/config/ApiPublicProperties.java]: Yeah, this is totally just a placeholder.

File: admin/pom.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (itzg) [reply admin/pom.xml]: Ah, I had fixed it in the admin pom but not the public one. Again, good catch.

**Comment by itzg** [comment]
@jjbuchan this is ready for re-review

**Pull request merged**